src/main/java/ti4/commands/custom/SpinTilesInRings.java

```diff
@@ -0,0 +1,133 @@
+package ti4.commands.custom;
+
+import java.util.ArrayList;
+import java.util.List;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import ti4.helpers.Constants;
+import ti4.helpers.DisplayType;
+import ti4.map.Game;
+import ti4.map.Tile;
+import ti4.message.MessageHelper;
+
+public class SpinTilesInRings extends CustomSubcommandData {
+    private static final String CW = "cw";
+    private static final String CCW = "ccw";
+
+    public SpinTilesInRings() {
+        super(Constants.SPIN_TILES_IN_RINGS, "Rotate the map according to fin logic or give custom rotations");
+        addOptions(new OptionData(OptionType.STRING, Constants.CUSTOM, "Custom rotation Ring:Direction:Steps 1:cw:1 2:ccw:2"));
+        addOptions(new OptionData(OptionType.STRING, Constants.MESSAGE, "Flavour message to send to main channel after spins"));
+    }
+
+    @Override
+    public void execute(SlashCommandInteractionEvent event) {
+        Game activeGame = getActiveGame();
+
+        if (event.getOption(Constants.CUSTOM) == null) {
+          spinRings(activeGame);
+        } else {
+          spinRingsCustom(activeGame, event);
+        }
+    }
+
+    public void spinRingsCustom(Game activeGame, SlashCommandInteractionEvent event) {
+      String[] customSpins = event.getOption(Constants.CUSTOM).getAsString().toLowerCase().split(" ");
+      List<Tile> tilesToSet = new ArrayList<>();
+      for (String spinString : customSpins) {
+
+        String[] spinSettings = spinString.split(":");
+        if (spinSettings.length != 3) {
+          MessageHelper.sendMessageToChannel(event.getMessageChannel(), "Invalid spin settings: " + spinString);
+          return;
+        }
+
+        int ring = Integer.parseInt(spinSettings[0]);
```

Comment:
I would throw these parseInts into a try/catch or some other validation that they are ints. Maybe just a regex pattern check to handle all validation